winter-framework/src/main/java/de/uni_mannheim/informatik/dws/winter/matching/rules/MatchingRule.java

```diff
@@ -66,80 +81,200 @@ public MatchingRule(double finalThreshold) {
+		if (position != -1) {
+			identifier = Integer.toString(position) + identifier;
+		}
+		Record debug = new Record(identifier);
+		debug.setValue(this.MATCHINGRULE, getClass().getSimpleName());
+		debug.setValue(this.RECORD1IDENTIFIER, record1.getIdentifier());
+		debug.setValue(this.RECORD2IDENTIFIER, record2.getIdentifier());
+
+		return debug;
+	}
+
+	public Record fillDebugRecord(Record debug, Comparator<RecordType, SchemaElementType> comperator, int position) {
+		Iterator<Attribute> schemaIterator = this.comparatorLog.getSchema().get().iterator();
+		ComparatorLogger compLog = comperator.getComparisonLog();
+		while (schemaIterator.hasNext()) {
+			Attribute schemaAtt = schemaIterator.next();
+			if (schemaAtt.getIdentifier()
+					.startsWith(Integer.toString(position) + '-' + comperator.getClass().getSimpleName())) {
+				Attribute compAtt = this.getComparatorToResultLog().get(schemaAtt);
+
+				if (compAtt == ComparatorLogger.RECORD1PREPROCESSEDVALUE) {
+					debug.setValue(schemaAtt, compLog.getRecord1PreprocessedValue());
+				} else if (compAtt == ComparatorLogger.RECORD2PREPROCESSEDVALUE) {
+					debug.setValue(schemaAtt, compLog.getRecord2PreprocessedValue());
+				} else if (compAtt == ComparatorLogger.POSTPROCESSEDSIMILARITY) {
+					debug.setValue(schemaAtt, compLog.getPostprocessedSimilarity());
+				} else if (compLog.hasValue(compAtt)) {
+					debug.setValue(schemaAtt, compLog.getValue(compAtt));
+				}
+			}
 		}
+		return debug;
+	}
+
+	public void finalizeDebugRecord(Record debug, Double similarity) {
+		if(similarity != null){
+			debug.setValue(TOTALSIMILARITY, Double.toString(similarity));
+		}
+		this.comparatorLog.add(debug);
+	}
+
+	public void finalizeDebugRecordShort(RecordType record1, RecordType record2,
+			Comparator<RecordType, SchemaElementType> comperator, int position) {
+		Record debug = initializeDebugRecord(record1, record2, position);
+		Iterator<Attribute> schemaIterator = this.comparatorLogShort.getSchema().get().iterator();
+		ComparatorLogger compLog = comperator.getComparisonLog();
+		while (schemaIterator.hasNext()) {
+			Attribute schemaAtt = schemaIterator.next();
+
+			if (schemaAtt == ComparatorLogger.RECORD1PREPROCESSEDVALUE) {
+				debug.setValue(schemaAtt, compLog.getRecord1PreprocessedValue());
+			} else if (schemaAtt == ComparatorLogger.RECORD2PREPROCESSEDVALUE) {
+				debug.setValue(schemaAtt, compLog.getRecord2PreprocessedValue());
+			} else if (schemaAtt == ComparatorLogger.POSTPROCESSEDSIMILARITY) {
+				debug.setValue(schemaAtt, compLog.getPostprocessedSimilarity());
+			} else if (compLog.hasValue(schemaAtt)) {
+				debug.setValue(schemaAtt, compLog.getValue(schemaAtt));
+			}
+		}
+		// logger.info(this.comparatorLogShort.size());
+		// logger.info(debug.getIdentifier());
+		this.comparatorLogShort.add(debug);
+	}
 	
-		this.comparatorLog = result;
+	public void fillSimilarity(RecordType record1, RecordType record2, double similarity){

```

Comment:
finalizeDebugRecordShort and fillSimilarity do the same - please create one method with two overloads so this is clear